### PR TITLE
[refactor](executor)Refactor workload meta update to be

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes

Refactor workload meta update to be

### CODE REVIEW

File: gensrc/thrift/BackendService.thrift [hunk 1]
```diff
@@ -165,9 +165,8 @@ enum TTopicInfoType {
 }
 
 struct TopicInfo {
-    1: optional string topic_key
-    2: required TTopicInfoType topic_type
-    3: optional map<string, string> info_map
+    1: required TTopicInfoType topic_type
```
Comment (yiguolei) [inline gensrc/thrift/BackendService.thrift:168]: never use  required field

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

File: be/src/common/config.h [hunk 2]
```diff
@@ -1161,8 +1161,8 @@ DECLARE_mInt32(tablet_schema_cache_recycle_interval);
 DECLARE_mBool(exit_on_exception);
 
 // cgroup
-DECLARE_String(doris_cgroup_cpu_path);
-DECLARE_Bool(enable_cgroup_cpu_soft_limit);
+DECLARE_mString(doris_cgroup_cpu_path);
+DECLARE_mBool(enable_cgroup_cpu_soft_limit);
```
Comment (yiguolei) [inline be/src/common/config.h:1165]: why do we need this config?

File: gensrc/thrift/BackendService.thrift [hunk 3]
```diff
@@ -165,9 +165,8 @@ enum TTopicInfoType {
 }
 
 struct TopicInfo {
-    1: optional string topic_key
-    2: required TTopicInfoType topic_type
-    3: optional map<string, string> info_map
+    1: optional TTopicInfoType topic_type
+    2: optional map<string, string> info_map
```
Comment (yiguolei) [inline gensrc/thrift/BackendService.thrift:169]: 我感觉我们用string 传递不是很好，

Comment (yiguolei) [reply gensrc/thrift/BackendService.thrift:169]: struct TWorkloadGroupInfo {
1: optional int cpu_soft_limit
2: optional int cpu_hard_limit
3: optional int mem_limit
4: optional bool mem_over_commit
}
struct TTopicInfo {
1: optional TTopicInfoType topic_type
2. optional WorkloadGroupInfo workload_group_info
}

File: be/src/runtime/fragment_mgr.cpp [hunk 4]
```diff
@@ -654,45 +654,28 @@ Status FragmentMgr::_get_query_ctx(const Params& params, TUniqueId query_id, boo
...
+                    std::stringstream ss;
+                    ss << "Query/load id: " << print_id(query_ctx->query_id());
+                    ss << " use task group " << task_group_ptr->debug_string();
+                    if (tg_mgr->enable_cpu_soft_limit() && !config::enable_cgroup_cpu_soft_limit) {
+                        query_ctx->set_task_group(task_group_ptr);
+                        ss << ", cpu soft limit based doris sche";
                     } else {
-                        if (!config::enable_cgroup_cpu_soft_limit) {
-                            query_ctx->set_task_group(tg);
+                        bool ret = tg_mgr->set_task_sche_4_query_ctx(tg_id, query_ctx.get());
```
Comment (yiguolei) [inline be/src/runtime/fragment_mgr.cpp:667]: set_task_sche_4_query_ctx  ---> set_task_sche_for_query_ctx

**Comment by wangbo** [comment]
run buildall

File: be/src/runtime/fragment_mgr.cpp [hunk 5]
```diff
@@ -654,45 +654,28 @@ Status FragmentMgr::_get_query_ctx(const Params& params, TUniqueId query_id, boo
...
-                                tg_id, tg_name, task_group_info.cpu_hard_limit,
-                                task_group_info.cpu_share, _exec_env, query_ctx.get());
-                        if (!ret.ok()) {
-                            LOG(INFO) << "workload group init failed "
-                                      << ", name=" << tg_name << ", id=" << tg_id
-                                      << ", reason=" << ret.to_string();
-                        }
+                uint64_t tg_id = params.workload_groups[0].id;
+                auto* tg_mgr = _exec_env->task_group_manager();
+                if (auto task_group_ptr = tg_mgr->get_task_group_by_id(tg_id)) {
```
Comment (yiguolei) [inline be/src/runtime/fragment_mgr.cpp:659]: 下面这堆ss 的代码，感觉就是  task group to string啊，直接在task group 里写一个方法，然后保存就行了，不用每次都生成

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by hf200012 — APPROVED** [review]

**Comment by hf200012** [comment]
run buildall

**Pull request merged**